### PR TITLE
Use a stack to store the trees/input while applying command edits

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

    Command edits are performed when there is a need to rewrite part of the
    user input, in order to support things like command substitution and
    also rewriting one command in terms of another (e.g. @@c: is rewritten
    in terms of @@=).
    
    It can happen that multiple nested edits are necessary in execute a
    script, thus it is necessary to maintain a stack of saved things,
    instead of just one element.

**Test plan**

Create a file with:
```
db @ main
db- @@c:dblq
dbil

```
then run `rizin -elog.level=1 -Q -i /tmp/in /bin/ls`. It affects 0.3.0 as well.
Test is going to come with the `db`-port PR.

**Closing issues**

...

### CODE REVIEW

File: librz/core/cmd.c [hunk 1]
```diff
@@ -3554,9 +3554,9 @@ struct tsr2cmd_state {
 	TSParser *parser;
 	RzCore *core;
 	char *input;
-	char *saved_input;
+	RzPVector/*<char *>*/ saved_input;
```
Comment (DMaroo) [inline librz/core/cmd.c]:
```suggestion
	RzPVector /*<char *>*/ saved_input;
```

File: librz/core/cmd.c [hunk 2]
```diff
@@ -3554,9 +3554,9 @@ struct tsr2cmd_state {
...
 	TSTree *tree;
-	TSTree *saved_tree;
+	RzPVector/*<TSTree *>*/ saved_tree;
```
Comment (DMaroo) [inline librz/core/cmd.c]:
```suggestion
	RzPVector /*<TSTree *>*/ saved_tree;
```

**Review by DMaroo — COMMENTED** [review]
To make `clang-format` happy.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Review by DMaroo — APPROVED** [review]
Please merge :)

**Pull request merged**